File: android/ssabree/app/src/main/java/com/ssafy/ssabree/core/datasource/local/AuthDataStore.kt
package com.ssafy.ssabree.core.datasource.local

import com.ssafy.ssabree.core.utils.SecureStorage
import com.ssafy.ssabree.core.utils.model.AuthTokens

class AuthDataStore(
    private val secureStorage: SecureStorage
) {
    fun isLoggedIn(): Boolean {
        return secureStorage.getTokens() != null
    }

    fun getAccessToken(): String? {
        return secureStorage.getTokens()?.accessToken
    }

    fun getTokenType(): String {
        return secureStorage.getTokens()?.tokenType ?: "Bearer"
    }

    fun getUId(): Int? {
        return secureStorage.getTokens()?.uid
    }

    fun getRefreshToken(): String? {
        return secureStorage.getTokens()?.refreshToken
    }

    suspend fun saveTokens(tokens: AuthTokens) {
        secureStorage.saveTokens(tokens)
    }

    suspend fun clear() {
        secureStorage.clearTokens()
    }
}

File: iOS/ssabree/ssabree/Core/Storage/AuthDataStore.swift
import Foundation

/// Auth data store that manages token storage and retrieval
/// Works with any SecureStore implementation (Keychain or InMemory)
final class AuthDataStore {
    private let secureStore: SecureStore

    init(secureStore: SecureStore) {
        self.secureStore = secureStore
    }

    // MARK: - Login State

    func isLoggedIn() -> Bool {
        getAccessToken() != nil
    }

    // MARK: - Token Access

    func getAccessToken() -> String? {
        secureStore.getTokens()?.accessToken
    }

    func getRefreshToken() -> String? {
        secureStore.getTokens()?.refreshToken
    }

    func getTokenType() -> String {
        secureStore.getTokens()?.tokenType ?? "Bearer"
    }

    func getTokens() -> AuthTokens? {
        secureStore.getTokens()
    }

    // MARK: - User Info

    func getUserId() -> String? {
        secureStore.getTokens()?.userId
    }

    func getUid() -> Int? {
        secureStore.getTokens()?.uid
    }

    // MARK: - Token Management

    func saveTokens(_ tokens: AuthTokens) {
        secureStore.saveTokens(tokens)
    }

    func clear() {
        secureStore.clearTokens()
    }
}
